.cirrus.yml

```diff
@@ -1103,6 +1106,83 @@ artifacts_task:
           type: application/octet-stream
 
 
+win_installer_task:
+    name: "Verify Win Installer Build"
+    alias: win_installer
+    depends_on:
+      - alt_build
+    windows_container:
+        image: cirrusci/windowsservercore:2019
+    env:
+        PATH: "${PATH};C:\\ProgramData\\chocolatey\\bin"
+        CIRRUS_SHELL: powershell
+        CIRRUS_CLONE_DEPTH: 1
+        # Fake version, we are only testing the installer functions, so version doesn't matter
+        WIN_INST_VER: 9.9.9
+        CIRRUS_WORKING_DIR: "${CIRRUS_DEFAULT_WORK}"
+    install_script: |
+        # Update service is required for dotnet 3.5 (dep of wix)
+        Set-Service -Name wuauserv -StartupType "Manual"
+        choco install -y wixtoolset mingw golang archiver
+        if ($LASTEXITCODE -ne 0) {
+            Exit 1
+        }
+    main_script: |
+        # Powershell doesn't exit after
+        function CheckExit {
+            if ($LASTEXITCODE -ne 0) {
+                Exit $LASTEXITCODE
+            }
+        }
+        function DownloadFile {
+            param([string]$url, [string]$file, [Int]$retries=5, [Int]$delay=8)
+            $ProgressPreference = 'SilentlyContinue';
+            Write-Host "Downloading $url to $file"
+            For($i = 0;;) {
+                Try {
+                    Invoke-WebRequest -UseBasicParsing -ErrorAction Stop -Uri $url -OutFile $file
+                    Break
+                } Catch {
+                    if (++$i -gt $retries) {
+                        throw $_.Exception
+                    }
+                    Write-Host "Download failed - retrying:" $_.Exception.Response.StatusCode
+                    Start-Sleep -Seconds $delay
+                }
+            }
+        }
+        # Drop global envs which have unix paths, defaults are fine
+        Remove-Item Env:\GOPATH
+        Remove-Item Env:\GOSRC
+        Remove-Item Env:\GOCACHE
+
+        Set-Location contrib\win-installer
```

Comment:
ROFL so true. They actually do have a shortcut for "cd" but it files warnings at you. Even more annoying when you define your own `cmdlets` they yell if you dont follow the approved noun and verb list: https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.2

Reply:
Wow. I'm sure glad you're on top of it and not me :grin:

Reply:
BTW in case it's useful for debugging / verifying what gets added to cirrus windows tasks, you can install powershell on Fedora they have an rpm and tarbal on their gh site: https://github.com/PowerShell/PowerShell/releases/tag/v7.2.6

It's not a 100% mirror since some scripts will use the Windows SDK/dotnet apis, but its close enough to validate.